nvdaHelper/localWin10/oneCoreSpeech.h

```diff
@@ -14,28 +14,45 @@ This license can be found at:
 */
 
 #pragma once
-#define export __declspec(dllexport) 
+#define export __declspec(dllexport)
 
-typedef void (*ocSpeech_Callback)(byte* data, int length, const char16* markers);
-typedef struct {
-	Windows::Media::SpeechSynthesis::SpeechSynthesizer ^synth;
+typedef void (*ocSpeech_Callback)(byte* data, int length, const wchar_t* markers);
+
+class OcSpeech {
+private:
+	winrt::Windows::Media::SpeechSynthesis::SpeechSynthesizer synth{ nullptr };
 	ocSpeech_Callback callback;
-} OcSpeech;
+
+public:
+	OcSpeech();
+	winrt::fire_and_forget speak(winrt::hstring text);

```

Comment:
One thing to watch, `fire_and_forget` [swallows exceptions ](https://devblogs.microsoft.com/oldnewthing/20190320-00/?p=102345) prior to `version 1901118.3` and terminates on exception after.

Reply:
I believe we catch everything we caught before, so this shouldn't hurt us I believe.

Reply:
You'll have to confirm which behaviour we get. If we are using post `version 1901118.3` then an exception will kill the app (via a call to terminate). Or perhaps you are saying that inside `speak` all exceptions are caught?

Reply:
Perhaps move the contents of `speak` into a private helper, then in the public `fire_and_forget` wrapper catch all exceptions and log them.

Reply:
I'm not sure whether I follow you correctly. Moving the contents of speak to a private helper still requires it to be an coroutine. Wouldn't it be enough to wrap just everything in a try block?